smart-hadoop-support/smart-hadoop/src/main/java/org/smartdata/hdfs/metric/fetcher/DataNodeInfoFetcher.java

```diff
@@ -103,8 +99,27 @@ public void run() {
       try {
         final List<DatanodeStorageReport> reports = getDNStorageReports();
         metaStore.deleteAllDataNodeInfo();
-        for(DatanodeStorageReport r : reports){
+        for(DatanodeStorageReport r : reports) {
           metaStore.insertDataNodeInfo(transform(r.getDatanodeInfo()));
+          //insert record in DataNodeStorageInfoTable
+          for (int i = 0; i < r.getStorageReports().length; i++) {
+            long sid = CompatibilityHelperLoader.getHelper().getSidInDatanodeStorageReport(
+                r.getStorageReports()[i].getStorage());
```

Comment:
Replace 'r.getStorageReports()[i]' with a temp variable, as it will be heavily referenced.